jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/xloader/ProcNodeTableBuilderX.java

```diff
@@ -99,22 +104,27 @@
         Process proc2;
         try {
             //LOG.info("Step : external sort");
-            String[] sortCmd =
-                { "sort",
+            List<String> sortCmdBasics = Arrays.asList(
+                  "sort",
                     "--temporary-directory="+loaderFiles.TMPDIR, "--buffer-size=50%",
                     "--parallel=2", "--unique",
-                    // Don't compress. There will be enough space because later we work on indexes.
                     //"--compress-program=/usr/bin/gzip",
                     "--key=1,1"
-                };
-            if ( sortArgs != null ) {}
+                    );
+
+            List<String> sortCmd = new ArrayList<>(sortCmdBasics);
 
+            //if ( sortNodeTableArgs != null ) {}
+
+            // See javadoc for CompressSortNodeTableFiles - usually false
+            if ( BulkLoaderX.CompressSortNodeTableFiles )
+                sortCmd.add("--compress-program=/usr/bin/gzip");
```

Comment:
Again just use `gzip`?

Also this should mean that the commented out line 111 can be deleted